src/mpi/init/mpir_finalize.c

```diff
@@ -66,11 +66,23 @@ static void MPIR_Call_finalize_callbacks(int min_prio, int max_prio)
 }
 
 int MPIR_Finalize_impl(void)
+{
+    return MPII_Finalize(NULL);
+}
+
+int MPII_Finalize(MPIR_Session * session_ptr)
 {
     int mpi_errno = MPI_SUCCESS;
     int rank = MPIR_Process.comm_world->rank;
+    int is_world_model = (session_ptr == NULL);
 
     MPIR_INIT_LOCK;
+
+    if (session_ptr) {
+        /* handle any clean up on session */
+        MPIR_Handle_obj_free(&MPIR_Session_mem, session_ptr);
+    }
+
```

Comment:
Since we have already defined `is_world_model`, can we check `if(! is_world_model )` instead of `if(session_ptr)`?